feat(HTTP): Refresh 3XX response pages (#35353)
mdn/content: files/en-us/web/http/status/307/index.md

{{HTTPSidebar}}

The HTTP **`307 Temporary Redirect`** [redirection response](/en-US/docs/Web/HTTP/Status#redirection_messages) status code indicates that the resource requested has been temporarily moved to the URL in the {{HTTPHeader("Location")}} header.

> A browser redirects automatically to the new page, but the ranking value in search engines (colloquially named "link juice" in {{Glossary("SEO")}} terms) is **not sent** to the new URL.

Reviewer: See my previous comments about search engines

The method and the body of the original request are reused to perform the redirected request.
In the cases where you want the request method to be changed to {{HTTPMethod("GET")}}, use {{HTTPStatus("303", "303 See Other")}} instead.
This is useful when you want to give an answer to a successful {{HTTPMethod("PUT")}} request that is not the uploaded resource, but a status monitor or confirmation message like "You have successfully uploaded XYZ".

The difference between `307` and {{HTTPStatus("302")}} is that `307` guarantees that the client **will not change** the request method and body when the redirected request is made.
With `302`, older clients incorrectly changed the method to {{HTTPMethod("GET")}}.
`307` and `302` responses are identical when the request method is {{HTTPMethod("GET")}}.

## Status

```http
307 Temporary Redirect
```

## Examples

### 307 response to a moved resource

> The following {{HTTPMethod("GET")}} request to is made to a resource with a `307` redirection in place.
> The {{HTTPHeader("Location")}} header shows the new URL in place for the resource.

Reviewer: I can see what you mean by "in place" but it feels repetetive.

```suggestion
The following {{HTTPMethod("GET")}} request to is made to a resource that has a `307` redirection in place.
The {{HTTPHeader("Location")}} header shows URL of the redirected resource.
```

Author: Good cleanup 👍🏻
